sycl/doc/extensions/DataFlowPipes/data_flow_pipes.asciidoc

```diff
@@ -170,6 +170,14 @@ The pipe class exposes static member functions for writing a data word to a pipe
 
 Blocking and non-blocking forms of the read and write members are defined, with the form chosen based on overload resolution.
 
+Read/write methods take a property_list argument, which can contain the latency control properties `latency_anchor_id` and/or `latency_constraint`.
+
+* `sycl::ext::intel::latency_anchor_id<N>`, where `N` is an integer: An ID to associate with the current read/write function call, which can then be referenced by other `latency_constraint` properties elsewhere in the program to define relative latency constaints. ID must be unique within the application, and a diagnostic is required if that condition is not met.
+* `sycl::ext::intel::latency_constraint<A, B, C>`: A tuple of three values which cause the current read/write function call to act as an endpoint of a latency constraint relative to a specified `latency_anchor_id` defined by a different instruction.
+** `A` is an integer: The ID of the target anchor defined on a different instruction through a `latency_anchor_id` property.
+** `B` is an enum value: The type of control from the set {`latency::exact`, `latency::min`, `latency::max`}.
```

Comment:
It's unclear where this enumeration resides.  Is it in the `ext::intel` namespace or a member of the `ext::intel::latency_constraint` property?  Maybe @rolandschulz or @Pennycook have a suggestion?  It would be good to have a consistent style for properties that take an enumeration as a template parameter.

Once we decide, the code synopsis below should be updated to show the definition of the property and this enumeration.

Reply:
I prefer to have the enumeration under `ext::intel` which simplifies/shortens the use model much more than the other way.

Reply:
I see advantages either way.  However, I'd like to choose a common style for properties that take an enum value.  For the proposed [device global spec](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/DeviceGlobal/SYCL_INTEL_device_global.asciidoc), we chose the style where the enum is inside the property struct:

```
struct host_access {
  enum class access: /*unspecified*/ {
    read,
    write,
    read_write,
    none
  };
  template<access A>
  using value_t = property_value<host_access, std::integral_constant<access, A>>;
};

struct init_mode {
  enum class trigger: /*unspecified*/ {
    reprogram,
    reset
  };
  template<trigger T>
  using value_t = property_value<init_mode, std::integral_constant<trigger, T>>;
};
```

That spec is still in the proposal stage, so we can change it if we prefer the other style.

As you point out @shuoniu-intel, defining the `enum` outside of the property struct is less verbose.  However, there is greater chance of conflict with enumeration names from other properties.  For example, that names `access` and `trigger` from the device global spec might be too generic and have a danger of colliding.  The enumeration name `latency` in this spec might also be in danger of colliding with another enumeration.

I think I have a slight preference for the style used by the device global spec because it reduces the chance of conflict and because it creates a clear association between the property and the enum.  I could be convinced otherwise, though if others feel differently.

A third option is to use an **unscoped** enumeration that is declared inside the property structure.  This is less verbose and still avoids conflict with enumerations from other properties:

```
struct host_access {
  enum access {
    read,
    write,
    read_write,
    none
  };
  template<access A>
  using value_t = property_value<host_access, std::integral_constant<access, A>>;
};
```

Maybe this would be a good compromise?

Reply:
Good point. I'm good with having the enum inside the property struct if it is the conventional and consistent way. 
I'll go for scoped enum class just like device global.